### PR TITLE
DM-28455: Cycle 17 support

### PR DESCRIPTION
**Pull request by tribeiro**
Update lsstsqre build to receive ts_idl version
Update version of the lsstsqre stack to w_2020_02.
Update pointing component to v2.4.0 and build scripts.
Update MTAOS version and build script. This latest version does not require lsst_sims anymore.
Update ScriptQueue build with additional dependencies for CWFS and rapid analysis scripts.
Update Scheduler version.

### CODE REVIEW

File: cycle/cycle.env [hunk 1]
```diff
@@ -30,19 +30,19 @@ GID=73006
...
 dds_community_version=6.9.0
 dds_community_build_el6=2.el6
-dds_community_build_el7=8
+dds_community_build_el7=12
 ts_dds_community=6.9.190705.post8
 dds_private_version=6.10.4
 dds_private_build_el6=1
 dds_private_build=6.el7
-ts_dds_private=6.10.4_6
+ts_dds_private=6.10.4
```
Comment (mareuter) [inline cycle/cycle.env:45]: Did something change in how the package is being versioned?

**Review by mareuter — COMMENTED** [review]
Would like to get some feedback on my questions before giving final approval.

File: build/conda_builder/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,20 @@
+ARG image_tag=7-stack-lsst_distrib-w_2021_04
```
[the remark this answers is not in the document]
Comment (tribeiro) [reply build/conda_builder/Dockerfile:1]: yes

File: build/mtaos_sim/Dockerfile [hunk 3]
```diff
@@ -39,18 +32,66 @@ RUN source ${WORKDIR}/loadLSST.bash && \
...
+    && /home/saluser/.checkout_repo.sh ${ts_ofc} \
+    && eups declare -r . ts_ofc ${ts_ofc} -t current \
+    && cd ${AOS_REPOS}/ts_MTAOS/ \
     && /home/saluser/.checkout_repo.sh ${mtaos} \
     && eups declare -r . ts_MTAOS ${mtaos} -t current \
     && cd ../ts_config_mttcs \
     && /home/saluser/.checkout_repo.sh ${ts_config_mttcs} \
     && eups declare -r . ts_config_mttcs ${ts_config_mttcs} -t current
 
+# Run scons on each package
```
[the remark this answers is not in the document]
Comment (tribeiro) [reply build/mtaos_sim/Dockerfile:58]: Problem is, we don't distribute these packages with eups... so when you want to "declare" a package inline you need to build them with scons again. some of the MTAOS dependencies in particular have C code that needs compiling.

File: build/ptg/Dockerfile [hunk 4]
```diff
@@ -86,9 +85,13 @@ RUN mkdir -p $PTG_DATA_DIR && cp $LSST_TOP/src/data/* $PTG_DATA_DIR/
 
 COPY ptg/ATPtg.info /home/saluser/repos/ts_pointing_common/install/data/
 COPY ptg/MTPtg.info /home/saluser/repos/ts_pointing_common/install/data/
+COPY ptg/at.mod /home/saluser/repos/ts_pointing_common/install/data/
```
[the remark this answers is not in the document]
Comment (tribeiro) [reply build/ptg/Dockerfile:88]: yes... you are right... Probably what we want to do is create a directory for ATPtg in ts_config_attc and put all this configuration in there. I'll have to handle this in a separate ticket.

File: build/ptg/startup.sh [hunk 5]
```diff
@@ -5,22 +5,16 @@ source ${HOME}/.setup_sal_env.sh
...
     exit 1
 fi
 
 wait ${pid}
-
-exit_code=$?
-
-echo ${TELESCOPE}Ptg exited with code ${exit_code}
-
-exit 0
```
[the remark this answers is not in the document]
Comment (tribeiro) [reply build/ptg/startup.sh:26]: yes they should.

File: build/scriptqueue/Dockerfile [hunk 6]
```diff
@@ -11,16 +11,19 @@ WORKDIR  ${SAL_HOME}
 RUN git clone https://github.com/lsst-ts/ts_config_attcs.git && \
     git clone https://github.com/lsst-ts/ts_config_ocs.git && \
     git clone https://github.com/lsst-ts/ts_observatory_control.git && \
+    git clone https://github.com/bxin/cwfs.git && \
```
[the remark this answers is not in the document]
Comment (tribeiro) [reply build/scriptqueue/Dockerfile:14]: Agree... in fact this is part of ts_wep but we will need to make some work to replace the things we have using Bo's repo to use the code in ts_wep.

File: cycle/cycle.env (hunk 1 above)
Comment (tribeiro) [reply cycle/cycle.env:45]: No, I build py38 and it had this tag in it. I need to roll back to `4_6`

File: build/deploy-env/conda/Dockerfile [hunk 7]
```diff
@@ -55,16 +55,17 @@ USER saluser
...
 ARG ts_idl
+ARG ts_salobj
 
 RUN curl https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh -o miniconda.sh && \
     chmod +x miniconda.sh && ./miniconda.sh -b -p ${WORKDIR}/miniconda && \
     source /home/saluser/.setup_sal_env.sh && \
     conda config --add channels conda-forge && \
     conda install -y cython && \
-    conda install -y -c lsstts -c lsstts/label/dev python=3.7 ts-salobj=${ts_salobj} ts-dds=${ts_dds} ts-idl=${ts_idl} && \
+    conda install -y -c lsstts -c lsstts/label/dev python=3.7 ts-ddsconfig=${ts_ddsconfig} ts-salobj=${ts_salobj} ts-dds=${ts_dds} ts-idl=${ts_idl} && \
```
[the remark this answers is not in the document]
Comment (tribeiro) [reply build/deploy-env/conda/Dockerfile:68]: When we tested SalObj 6 at NCSA long ago the packages where labeled as dev. So, I needed this here to be able to install them. I am cross between removing it as we may need in the future if we ever want to test a dev version of any of these packages again.

File: build/deploy-env/lsstsqre/Dockerfile [hunk 8]
```diff
@@ -64,16 +64,21 @@ RUN source ${WORKDIR}/.setup_sal_env.sh && \
...
 ARG ts_dds
+ARG ts_ddsconfig
+ARG ts_idl
+ARG ts_salobj
 
 RUN source ${WORKDIR}/.setup_sal_env.sh && \
     conda config --add channels conda-forge && \
     conda install -y cython && \
-    conda install -y -c lsstts -c lsstts/label/dev ts-dds=${ts_dds} ts-salobj=${ts_salobj} ts-idl && \
+    conda install -y -c lsstts -c lsstts/label/dev ts-dds=${ts_dds} ts-ddsconfig=${ts_ddsconfig} ts-salobj=${ts_salobj} ts-idl=${ts_idl} && \
```
[the remark this answers is not in the document]
Comment (tribeiro) [reply build/deploy-env/lsstsqre/Dockerfile:75]: same answer 😉

File: build/mtaos_sim/Dockerfile (hunk 3 above)
[the remark this answers is not in the document]
Comment (mareuter) [reply build/mtaos_sim/Dockerfile:58]: Have we thought about using the stack binary distribution to get around some of the compiling?

**Review by mareuter — APPROVED** [review]
Thanks for the feedback. PR approved.

File: build/mtaos_sim/Dockerfile (hunk 3 above)
[the remark this answers is not in the document]
Comment (tribeiro) [reply build/mtaos_sim/Dockerfile:58]: I've been thinking about it but we can't really rely on the stack binary distribution for our own packages. But I am investigating the possibility of having our own eups distribution service (using nexus).

**Pull request merged**